client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -517,107 +521,29 @@ def inject_ayon_environment(deadlinePlugin):
-                " AYON_FOLDER_PATH, AYON_TASK_NAME, AYON_APP_NAME"
-            ))
-
-        # Use applications addon arguments
-        # TODO validate if applications addon should be used
-        args = [
-            "--headless",
-            "addon",
-            "applications",
-            "extractenvironments",
-            export_url
-        ]
-
-        # staging requires passing argument
-        # TODO could be removed when PR in ayon-core starts to fill
-        #  'AYON_USE_STAGING' (https://github.com/ynput/ayon-core/pull/1130)
-        #  - add requirement for "core>=1.1.1" to 'package.py' when removed
-        settings_variant = job.GetJobEnvironmentKeyValue(
-            "AYON_DEFAULT_SETTINGS_VARIANT"
-        )
-        if settings_variant == "staging":
-            args.append("--use-staging")
-
-        # Backwards compatibility for older versions
-        legacy_args = [
-            "--headless",
-            "extractenvironments",
-            export_url
-        ]
-
-        for key, value in add_kwargs.items():
-            args.extend(["--{}".format(key), value])
-            # Legacy arguments expect '--asset' instead of '--folder'
-            if key == "folder":
-                key = "asset"
-            legacy_args.extend(["--{}".format(key), value])
-
-        environment = {
-            "AYON_SERVER_URL": ayon_server_url,
-            "AYON_API_KEY": ayon_api_key,
-            "AYON_BUNDLE_NAME": ayon_bundle_name,
-        }
-
-        for key in ("AYON_USE_STAGING", "AYON_IN_TESTS"):
-            value = job.GetJobEnvironmentKeyValue(key)
-            if value:
-                environment[key] = value
-
-        for env, val in environment.items():
-            # Add the env var for the Render Plugin that is about to render
-            deadlinePlugin.SetEnvironmentVariable(env, val)
-            # Add the env var for current calls to `DeadlinePlugin.RunProcess`
-            deadlinePlugin.SetProcessEnvironmentVariable(env, val)
-
-        args_str = subprocess.list2cmdline(args)
-        print(">>> Executing: {} {}".format(exe, args_str))
-        process_exitcode = deadlinePlugin.RunProcess(
-            exe, args_str, os.path.dirname(exe), -1
-        )
-
-        if process_exitcode != 0:
-            print(
-                "Failed to run AYON process to extract environments. Trying"
-                " to use legacy arguments."
-            )
-            legacy_args_str = subprocess.list2cmdline(legacy_args)
-            process_exitcode = deadlinePlugin.RunProcess(
-                exe, legacy_args_str, os.path.dirname(exe), -1
-            )
-            if process_exitcode != 0:
-                raise RuntimeError(
-                    "Failed to run AYON process to extract environments."
+        output_dir = _get_output_dir(job)
+
+        worker_platform = platform.system().lower()
+        environment_file_name = f"extractenvironments_{worker_platform}.txt"
```

Comment:
Is this a `JSON` file? if so, suffix should be `json`
Also, I wonder if fully writing out `extractenvironments_` in the name is really worth it, why not just `env_windows.json`

ALSO, worse... is that this seems to FULLY ignore the fact that the resulting extracted environment is host specific, the AYON environments are build against your local environment (and appended to it).

Also, some environments may be using things like user profiles directories, including the paths like `C:\Users\<username>\...` etc. meaning that these are also dependent on host username.

Reply:
Note: I think the idea is/was that it happens only once per platform, but I don't think that can work.

So, I do 100% agree.

And also don't understand why `.txt` is extension of the file?

Reply:
BTW in that case it might worth to not store it to the render directory but e.g. to temp? (e.g. by some hash) -> not necessary now, but it would also resolve this https://github.com/ynput/ayon-deadline/pull/169/files#r2035546921 .

Reply:
Honestly - we should store the **NON-compiled** environment in the txt file. So just store the relevant application env json and the tools envs jsons. Then just run the 'build' logic of the environment locally - which doesn't need access to the database.

The downside there though is that we can't pre-run the prelaunch hooks. Because those can have any sort of custom logic.